### PR TITLE
TV: Fix error matching for ErrNoSuchEntity

### LINKED ISSUE #213: Broadcast: New broadcast fails to save
Since this change #190 we can't create new broadcasts since we are now using Update for calls to the broadcast even if it doesn't exist. This results in the save failing.

### PR DESCRIPTION
**Pull request by ao-david**
When saving a new broadcast, errors coming from cloudstore are now properly matched with the correct types from cloud.google.com/go/datastore.

This change also changes the updateConfig function since there will be no config to unmarshal

NOTE:
This is currently deployed to fix a P0 level bug

closes #213

### CODE REVIEW

File: cmd/oceantv/utils.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ import (
 	"strings"
 	"time"
 
+	googledatastore "cloud.google.com/go/datastore"
```
Comment (scruzin) [inline cmd/oceantv/utils.go:40]: Using "cloud.google.com/go/datastore" is breaking the encapsulation provided by openfish/datastore.
Please do this another way.

PS Apart from breaking the encapsulation, it breaks standalone mode.

File: cmd/oceantv/utils.go [hunk 2]
```diff
@@ -162,12 +163,39 @@ func updateConfigWithTransaction(ctx context.Context, store Store, skey int64, b
 	}
 
 	err := store.Update(ctx, key, updateConfig, &model.Variable{})
-	if errors.Is(err, datastore.ErrNoSuchEntity) {
+	// We need to check for both filestore and cloudstore error types here.
+	if errors.Is(err, datastore.ErrNoSuchEntity) || errors.Is(err, googledatastore.ErrNoSuchEntity) {
```
Comment (scruzin) [inline cmd/oceantv/utils.go:167]: Test for openfish/datastore errors, not cloud.google.com/go/datastore errors.

**Review by scruzin — CHANGES_REQUESTED** [review]

**Comment by ao-david** [comment]
After a little thinking I think what needs to happen is the openfish/datastore layer needs to check for the errNoExistError coming from google datastore, and replace it with a openfish error which we can then read from cloud.

File: cmd/oceantv/utils.go (hunk 2 above)
Comment (trekhopton) [reply cmd/oceantv/utils.go:167]: This can now be left just as datastore.ErrNoSuchEntity since this PR:
https://github.com/ausocean/openfish/pull/147

**Review by scruzin — APPROVED** [review]
LGTM. Please update the version too.

**Pull request merged**